### PR TITLE
Enhanced RDoc for Array

### PR DESCRIPTION
**Pull request by BurdetteLamar**
Methods:
- \+
- concat
- \*
- assoc
- rassoc

### CODE REVIEW

File: array.c [hunk 1]
```diff
@@ -5447,19 +5451,51 @@ rb_ary_concat(VALUE x, VALUE y)
...
+ *  - Argument +n+, if given, must be an
+ *    {Integer-convertible object}[doc/implicit_conversion_rdoc.html#label-Integer-Convertible+Objects].
+ *  - Argument +string_separator+, if given, myst be a
+ *    {String-convertible object}[doc/implicit_conversion_rdoc.html#label-String-Convertible+Objects].
  *
- *  Repetition --- With a String argument, equivalent to
- *  <code>ary.join(str)</code>.
+ *  ---
  *
- *  Otherwise, returns a new array built by concatenating the +int+ copies of
```
Comment (jeremyevans) [inline array.c:5456]: I think the older wording is more succinct and just as accurate, so I would keep it.    I would add your examples, though.

Comment (BurdetteLamar) [reply array.c:5456]: I think I've done what you suggested.  Squawk if not.

Comment (jeremyevans) [reply array.c:5456]: Due to the way you've moved things around, keeping the older wording would be replacing:

```
When argument +n+ is given, returns a new \Array built by concatenating +n+ copies of +self+.
If +n+ is positive, returns the concatenation of +n+ repetitions of +self+:
If +n+ is zero, returns an new empty \Array:
```

with: 

```
When argument +n+ is given, returns a new array built by concatenating the +int+ copies of +self+.
```

Comment (BurdetteLamar) [reply array.c:5456]: One more time.

Comment (jeremyevans) [reply array.c:5456]: Looks good, thank you.

**Comment by BurdetteLamar** [comment]
@jeremyevans this all doc (no code).  Think it's ok to merge with failing tests?

**Comment by jeremyevans** [comment]
You can probably rebase against current master, as earlier today the master branch had some issues.  It's best to use `[ci skip]` in your commit subject for pure documentation changes, though I'm not sure that actually skips all CI tests in pull requests (it's worth testing).

**Comment by BurdetteLamar** [comment]
> You can probably rebase against current master, as earlier today the master branch had some issues. It's best to use `[ci skip]` in your commit subject for pure documentation changes, though I'm not sure that actually skips all CI tests in pull requests (it's worth testing).

[ci skip] does not suppress testing, I've seen.  Don't know rebase.  Can I save my array.c, refork, reclone, restore array.c, add, commit, push?

**Comment by jeremyevans** [comment]
Assuming that `upstream` is the ruby upstream repository (the git remote), from your current branch:

```
git fetch upstream master
git rebase upstream/master
```

You can then push the result to your GitHub fork (`git push -f origin master` if `origin` is the git remote for your GitHub fork).

However, if you aren't sure about your git remotes and don't have a good understanding of the above commands, it should be fine to just use the `Rebase and merge` button on GitHub.

**Pull request merged**